Source/WebKit/UIProcess/PDF/WKPDFPageNumberIndicator.mm

```diff
@@ -0,0 +1,201 @@
+    [_label setOpaque:NO];
+    [_label setBackgroundColor:nil];
+    [_label setTextAlignment:NSTextAlignmentCenter];
+    [_label setFont:[UIFont boldSystemFontOfSize:indicatorFontSize]];
+    [_label setTextColor:[UIColor blackColor]];
+    [_label setAlpha:indicatorLabelOpacity];
+    [_label setAdjustsFontSizeToFitWidth:YES];
+    [_label setAutoresizingMask:UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight];
+    WebCore::PlatformCAFilters::setBlendingFiltersOnLayer([_label layer], WebCore::BlendMode::PlusDarker);
+    [[_backdropView contentView] addSubview:_label.get()];
+
+    [self updatePosition:self.frame];
+    [self setPageCount:pageCount];
+
+    return self;
+}
+
+- (void)dealloc
+{
+    [_timer invalidate];
+    [super dealloc];
+}
+
+- (void)updatePosition:(CGRect)frame
+{
+    RetainPtr webView = _webView.get();
+    if (!webView)
+        return;
+
+    UIEdgeInsets insets = UIEdgeInsetsAdd([webView _computedUnobscuredSafeAreaInset], [webView _computedObscuredInset], UIRectEdgeAll);
+    [self _moveToPoint:UIEdgeInsetsInsetRect(frame, insets).origin animated:NO completionHandler:[view = retainPtr(self)](bool originMoved) {
+        if (!originMoved)
+            return;
+        [view sizeToFit];
+        [view show];
+    }];
+}
+
+- (void)setCurrentPageNumber:(unsigned)currentPageNumber
+{
+    ASSERT(currentPageNumber);
+    bool labelTextShouldChange = _currentPageNumber != currentPageNumber;
+    if (labelTextShouldChange)
+        _currentPageNumber = currentPageNumber;
+    [self _updateLabel:labelTextShouldChange];
+}
+
+- (void)setPageCount:(unsigned)pageCount
+{
+    ASSERT(pageCount);
+    bool labelTextShouldChange = _pageCount != pageCount;
+    if (labelTextShouldChange)
+        _pageCount = pageCount;
+    [self _updateLabel:labelTextShouldChange];
+}
+
+- (void)show
+{
+    [UIView animateWithDuration:indicatorFadeInDuration.seconds() animations:[view = retainPtr(self)] {
+        [view setAlpha:1];
+    }];
+
+    if (_timer)
+        [_timer setFireDate:[NSDate dateWithTimeIntervalSinceNow:indicatorTimeout.seconds()]];
+    else
+        _timer = [NSTimer scheduledTimerWithTimeInterval:indicatorTimeout.seconds() target:self selector:@selector(hide:) userInfo:nil repeats:NO];
+}
+
+- (void)hide:(NSTimer *)timer
+{
+    [UIView animateWithDuration:indicatorFadeOutDuration.seconds() animations:[view = retainPtr(self)] {
+        [view setAlpha:0];
+    }];
+
+    [std::exchange(_timer, nil) invalidate];
+}
+
+- (void)_moveToPoint:(CGPoint)point animated:(BOOL)animated completionHandler:(WTF::CompletionHandler<void(bool)>&&)completionHandler
+{
+    point.x += indicatorMargin;
+    point.y += indicatorMargin;
+
+    auto animations = [view = retainPtr(self), point, completionHandler = WTFMove(completionHandler)] mutable {
+        CGRect frame = [view frame];
+        bool originMoved = static_cast<bool>(CGPointEqualToPoint(frame.origin, point) == NO);
```

Comment:
This `static_cast` is unnecessary. I'll remove it

Reply:
Yeah, I think this should be sufficient:

```suggestion
        bool originMoved = !CGPointEqualToPoint(frame.origin, point);
```

Reply:
Fixed!